admin-ui:build fails, property does not exist

### Bug Description

```
#17 25.59 admin-ui:build: ./src/features/datastore-connections/system_portal_config/forms/ConnectorParametersForm.tsx:295:19
#17 25.59 admin-ui:build: Type error: Type '{ connectionKey: string; onDelete: () => void; deleteResult: any; }' is not assignable to type 'IntrinsicAttributes & DataConnectionProps & { children?: ReactNode; }'.
#17 25.59 admin-ui:build:   Property 'connectionKey' does not exist on type 'IntrinsicAttributes & DataConnectionProps & { children?: ReactNode; }'.
#17 25.59 admin-ui:build: 
#17 25.59 admin-ui:build:   293 |               {connectionConfig ? (
#17 25.59 admin-ui:build:   294 |                 <DeleteConnectionModal
#17 25.59 admin-ui:build: > 295 |                   connectionKey={connectionConfig.key}
#17 25.59 admin-ui:build:       |                   ^
#17 25.59 admin-ui:build:   296 |                   onDelete={onDelete}
#17 25.59 admin-ui:build:   297 |                   deleteResult={deleteResult}
#17 25.59 admin-ui:build:   298 |                 />
#17 26.34 admin-ui:build: npm ERR! Lifecycle script `build` failed with error: 
#17 26.34 admin-ui:build: npm ERR! Error: command failed 
#17 26.34 admin-ui:build: npm ERR!   in workspace: admin-ui 
#17 26.35 admin-ui:build: npm ERR!   at location: /fides/clients/admin-ui 
#17 26.40 admin-ui:build: ERROR: command finished with error: command (/fides/clients/admin-ui) npm run build exited (1)
```

#### Steps to Reproduce

1. Run `nox -s "fides_env(test)"`

### Expected behavior

The build should run

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -44,6 +44,7 @@ The types of changes are:
 
 ### Fixed
 - Privacy notice UI's list of possible regions now matches the backend's list [#3787](https://github.com/ethyca/fides/pull/3787)
+- Admin UI "property does not existing" build issue [#3831](https://github.com/ethyca/fides/pull/3831)
 
 ## [2.16.0](https://github.com/ethyca/fides/compare/2.15.1...2.16.0)
 

--- a/clients/admin-ui/src/features/datastore-connections/system_portal_config/forms/ConnectorParametersForm.tsx
+++ b/clients/admin-ui/src/features/datastore-connections/system_portal_config/forms/ConnectorParametersForm.tsx
@@ -292,7 +292,6 @@ const ConnectorParametersForm: React.FC<ConnectorParametersFormProps> = ({
               ) : null}
               {connectionConfig ? (
                 <DeleteConnectionModal
-                  connectionKey={connectionConfig.key}
                   onDelete={onDelete}
                   deleteResult={deleteResult}
                 />

